### PR TITLE
GG-26684 Can't set serialized enum to a BinaryObject's field

### PR DESCRIPTION
**Pull request by korlov42**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/binary/builder/BinaryBuilderSerializer.java [hunk 1]
```diff
@@ -110,6 +111,16 @@ public void writeValue(BinaryWriterExImpl writer, Object val, boolean forceCol,
             return;
         }
 
+        if (val instanceof BinaryEnumObjectImpl) {
+            BinaryEnumObjectImpl obj = (BinaryEnumObjectImpl)val;
```
Comment (pavlukhin) [inline modules/core/src/main/java/org/apache/ignite/internal/binary/builder/BinaryBuilderSerializer.java]: Please check serialization for `GridBinaryMarshaller.UNREGISTERED_TYPE_ID`. There are some relevant code in `org.apache.ignite.internal.binary.builder.BinaryBuilderEnum#writeTo` and `org.apache.ignite.internal.binary.builder.BinaryEnumArrayLazyValue#writeTo`.

File: modules/core/src/test/java/org/apache/ignite/internal/binary/BinaryObjectBuilderAdditionalSelfTest.java [hunk 2]
```diff
@@ -1617,28 +1617,30 @@ public void testBuilderExternalizable() throws Exception {
...
+        BinaryObjectBuilder builder = newWrapper(TestClsWithEnum.class.getName());
 
         final TestEnum exp = TestEnum.A;
         final TestEnum[] expArr = {TestEnum.A, TestEnum.B};
 
         BinaryObject enumObj = builder.setField("testEnum", exp).setField("testEnumArr", expArr).build();
 
-        assertEquals(exp, ((BinaryObject)enumObj.field("testEnum")).deserialize());
-        Assert.assertArrayEquals(expArr, (Object[])deserializeEnumBinaryArray(enumObj.field("testEnumArr")));
+        Assert.assertEquals(exp, ((BinaryObject)enumObj.field("testEnum")).deserialize());
```
Comment (AMashenkov) [inline modules/core/src/test/java/org/apache/ignite/internal/binary/BinaryObjectBuilderAdditionalSelfTest.java]: Actually, assertSame is expected here.

File: modules/core/src/main/java/org/apache/ignite/internal/binary/builder/BinaryBuilderSerializer.java (hunk 1 above)
Comment (korlov42) [reply modules/core/src/main/java/org/apache/ignite/internal/binary/builder/BinaryBuilderSerializer.java]: thanks for noticing. I'll fix it.

File: modules/core/src/test/java/org/apache/ignite/internal/binary/BinaryObjectBuilderAdditionalSelfTest.java (hunk 2 above)
Comment (korlov42) [reply modules/core/src/test/java/org/apache/ignite/internal/binary/BinaryObjectBuilderAdditionalSelfTest.java]: done

**Pull request merged**